lib/auth/auth.go

```diff
@@ -3764,3 +3764,14 @@ func WithClusterCAs(tlsConfig *tls.Config, ap AccessCache, currentClusterName st
 		return tlsCopy, nil
 	}
 }
+
+// DefaultPrincipalsForRole returns default principals for the specified role.
+func DefaultPrincipalsForRole(role types.SystemRole) []string {
+	if (types.SystemRoles{role}).IncludeAny(types.RoleAuth, types.RoleAdmin, types.RoleProxy, types.RoleKube, types.RoleApp) {
```

Comment:
Should we include `types.RoleDatabase` and `types.RoleWindowsDesktop` here as well?

Reply:
The RoleDatabase and `RoleWindowsDesktop` uses reverse tunnel connection so APIHost is not needed because reverse tunnel dial is based hostID though I will add this for consistency.